### PR TITLE
Add contact pictures for email thread participants

### PR DESCRIPTION
**Pull request by heathergh**
# Changes

- Created a `ContactImage` component that accepts a contact object and will render either an image for the avatar or the contact's initials. If there is no contact, it will show the first letter of their name

- **Note:** Since there are significant design changes to the mailbox and email components, I am leaving that for another PR

## Before
Screen Shot 2021-08-12 at 11 01 39 AM

## After
Screen Shot 2021-08-12 at 10 54 14 AM

# Readiness checklist

- [ ] Cypress tests passing?
- [ ] New cypress tests added?
- [x] Included before/after screenshots, if the change is visual

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by ozsivanov — APPROVED** [review]
Looks good. Do we also want to use the `ContactImage` component in contact-list?

File: components/email/src/Email.svelte [hunk 1]
```diff
@@ -159,6 +189,36 @@
...
+  let contactQuery: ContactsQuery;
+  $: contactQuery = {
+    component_id: id,
+    access_token,
+  };
+
+  /*
+    Fetches contact for ContactImage component
+  */
+  async function getContact(account) {
```
Comment (AaronDDM) [inline components/email/src/Email.svelte:202]: Might be for another PR, but: I think we need to move this query method into a store and utilize the store for caching. Otherwise, we're going to be making multiple requests for the same email lookup multiple times.

File: components/email/src/Email.svelte [hunk 2]
```diff
@@ -55,6 +60,26 @@
...
+      contact = await getContact(
+        activeThread.messages[activeThread.messages.length - 1].from[
+          activeThread.messages[activeThread.messages.length - 1].from.length -
+            1
+        ],
+      );
+    } else if (message) {
+      contact = await getContact(message.from[message.from.length - 1]);
+    }
+  })();
```
Comment (AaronDDM) [inline components/email/src/Email.svelte:81]: We'll not want to call this if no `component_id` is provided, since these requests will always fail.

File: commons/src/components/ContactImage/ContactImage.svelte [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  export let contact;
+  export let contactQuery;
+  export let height = "32px";
+  export let width = "32px";
+
+  let image;
+
+  beforeUpdate(async () => {
+    if (contact?.picture_url) {
+      image = await fetchContactImage(contactQuery, contact?.id);
```
Comment (AaronDDM) [inline commons/src/components/ContactImage/ContactImage.svelte:15]: This maybe beyond the scope of this PR, but we'll probably also want to cache the results from our queries into the store and fetch from the store if we are making the same lookup again.

**Review by AaronDDM — CHANGES_REQUESTED** [review]
@heathergh we need to figure out what is going on with this implementation in the Mailbox component. Something is causing a lot of requests to be made.

File: commons/src/components/ContactImage/ContactImage.svelte (hunk 3 above)
Comment (heathergh) [reply commons/src/components/ContactImage/ContactImage.svelte:15]: I'll make this another story for next week

**Review by wonka-lighthouse — COMMENTED** [review]
| Device | URL |
|--|--|
| mobile | http://localhost:8000/components/agenda/src/index.html |

     
| Device | URL |
|--|--|
| mobile | http://localhost:8000/components/composer/src/index.html |

Not what you expected? Are your scores flaky? [Run Lighthouse on Foo](https://www.foo.software/lighthouse)
If scores continue to be inconsistent consider [running all audits on Foo](https://github.com/foo-software/lighthouse-check-action#usage-foos-automated-lighthouse-check-api)

File: components/email/src/Email.svelte [hunk 4]
```diff
@@ -55,6 +61,28 @@
     ]) || {}) as EmailProperties;
   });
 
+  let contact;
+  $: (async () => {
+    if (show_contact_avatar) {
```
Comment (AaronDDM) [inline components/email/src/Email.svelte:66]:
```suggestion
    if (!contact) {
```

File: components/email/src/Email.svelte [hunk 5]
```diff
@@ -95,6 +123,11 @@
       "default",
     );
     is_starred = getPropertyValue(internalProps.is_starred, is_starred, false);
+    show_contact_avatar = getPropertyValue(
+      internalProps.show_contact_avatar,
+      show_contact_avatar,
+      false, // setting this to false for now, so mailbox can load
```
Comment (AaronDDM) [inline components/email/src/Email.svelte:129]:
```suggestion
      true,
```

**Review by AaronDDM — APPROVED** [review]

File: commons/src/store/contacts.ts [hunk 6]
```diff
@@ -15,6 +22,28 @@ function initializeContacts() {
         return contacts;
       });
     },
+    addContact: async (query: ContactSearchQuery) => {
```
Comment (AaronDDM) [inline commons/src/store/contacts.ts:25]: Lovely!
